libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/NavigationView.kt

```diff
@@ -298,145 +424,169 @@ class NavigationView : ConstraintLayout {
                     routeArrowViewModel.routeProgressUpdated(routeProgress, style)
                 }
                 externalRouteProgressObservers.forEach {
-                    lifeCycleOwner.lifecycleScope.launch {
-                        it.onRouteProgressChanged(routeProgress)
+                    it.onRouteProgressChanged(routeProgress)
+                }
+                uiComponents.forEach { uiComponent ->
+                    when (uiComponent) {
+                        is ManeuverUIComponent.MapboxManeuverUIComponent -> {
+                            uiComponent.onRouteProgressChanged(routeProgress)
+                        }
+                        is TripProgressUIComponent.MapboxTripProgressUIComponent -> {
+                            uiComponent.onRouteProgressChanged(routeProgress)
+                        }
                     }
                 }
             }
         }
+    }
 
+    private fun observeLocationMatcherResults() {
         lifeCycleOwner.lifecycleScope.launch {
-            mapboxNavigationViewModel.routesUpdatedResults.collect { routesUpdatedResult ->
-                // view models that need route progress updates should be added here
-                mapView.getMapboxMap().getStyle()?.let { style ->
-                    routeLineViewModel.routesUpdated(routesUpdatedResult, style)
+            mapboxNavigationViewModel.newLocationMatcherResults.collect { locationMatcherResult ->
+                externalLocationObservers.forEach {
+                    it.onNewLocationMatcherResult(locationMatcherResult)
                 }
-                externalRoutesObservers.forEach {
-                    lifeCycleOwner.lifecycleScope.launch {
-                        it.onRoutesChanged(routesUpdatedResult)
+                uiComponents.forEach { uiComponent ->
+                    when (uiComponent) {
+                        is SpeedLimitUIComponent.MapboxSpeedLimitUIComponent -> {
+                            uiComponent.onNewLocationMatcherResult(locationMatcherResult)
+                        }
                     }
                 }
+                navigationLocationProvider.changePosition(
+                    locationMatcherResult.enhancedLocation,
+                    locationMatcherResult.keyPoints
+                )
```

Comment:
We should have a `component` to manage the puck and wire these operations through that `component`.